packages/react-aria-components/src/ComboBox.tsx

```diff
@@ -22,6 +22,17 @@ import React, {createContext, ForwardedRef, forwardRef, useCallback, useMemo, us
 import {TextContext} from './Text';
 import {useCollection} from './Collection';
 
+export interface ComboBoxRenderProps {
+  /**
+   * Whether the combobox is focused, either via a mouse or keyboard.
+   */
+  isFocused: boolean,
+  /**
+   * Whether the combobox is currently open.
+   */
+  isOpen: boolean
```

Comment:
Not entirely sure if it is worthwhile adding `data-*` selectors onto the outermost div that implementors can target, open to opinions.